torch/custom_class.h

```diff
@@ -263,11 +298,46 @@ class class_ {
 
  private:
   template <typename Func>
-  void defineMethod(std::string name, Func func, std::string doc_string = "") {
+  void defineMethod(
+      std::string name,
+      Func func,
+      std::string doc_string = "",
+      std::initializer_list<arg> default_args = {}) {
```

Comment:
Nit:

Could we consider changing the term `default argument` to `keyword argument` or simply `arg` throughout the PR? the structure `arg` isn't really about default values, they describe all aspects of an argument. IMHO, It would be much clearer.

Reply:
https://pybind11.readthedocs.io/en/stable/basics.html#default-args

The term "default argument" is used here in the pybind documentation

Reply:
`py::arg` is similarly used in more than just expressing default argument, see here:
https://pybind11.readthedocs.io/en/stable/basics.html#keyword-arguments

But this is really minor, up to you.